h/static/scripts/tests/controllers/share-widget-controller-test.js

```diff
@@ -0,0 +1,144 @@
+'use strict';
+
+const syn = require('syn');
+
+const ShareWidgetController = require('../../controllers/share-widget-controller');
+
+describe('ShareWidgetController', () => {
+
+  const widgetTemplate = `
+  <div class="js-share-widget">
+    <div class="share-widget">
+      <div class="share-widget-target">
+        <a share-target-href="https://twitter.com/intent/tweet?url={href}&hashtags=annotated"
+          class="share-widget-target__icon">
+          {{ svg_icon('twitter', 'share-widget-action') }}
+        </a>
+        <a share-target-href="https://www.facebook.com/sharer/sharer.php?u={href}"
+          class="share-widget-target__icon">
+          {{ svg_icon('facebook', 'share-widget-action') }}
+        </a>
+        <a share-target-href="https://plus.google.com/share?url={href}"
+          class="share-widget-target__icon">
+          {{ svg_icon('google-plus', 'share-widget-action') }}
+        </a>
+        <a share-target-href="mailto:?subject=Let's%20Annotate&amp;body={href}"
+          class="share-widget-target__icon">
+        </a>
+      </div>
+      <div class="share-widget-link">
+          <input class="share-widget-link__text value="">
+      </div>
+      <div class="share-widget-msg share-widget-msg-group">GROUP</div>
+      <div class="share-widget-msg share-widget-msg-private">PRIVATE</div>
+    </div>
+  </div>
+  `;
+
+  const template = `
+    <button share-widget-config='{"url":"url1", "private":false, "group":false}'>
+      btn
+    </button>
+    <button share-widget-config='{"url":"url2", "private":true, "group":false}'>
+      btn
+    </button>
+    <button share-widget-config='{"url":"url3", "private":false, "group":true}'>
+      btn
+    </button>
+    <button share-widget-config='{"url":"url4", "private":true, "group":true}'>
+      <span><b>btn</b></span>
+    </button>
+  `;
+
+  let container;
+  let ctrl;
+
+  const getTriggers = () => {
+    return Array.from(container.querySelectorAll('[share-widget-config]'));
+  };
+
+  const widgetIsVisble = () => {
+    return container.querySelector('.js-share-widget').style.visibility === 'visible';
+  };
+
+  const urlsReflectUpdate = (addedValue) => {
```

Comment:
It wasn't that clear to me from the function and argument names here what this does. `shareLinksMatchUrl = (url) => {...}`?